test/js/node/fs/fs-mkdir.test.ts

```
@@ -0,0 +1,346 @@
+    });
+  });
+});
+
+describe("fs.mkdir - return values", () => {
+  let tmpdir: string;
+
+  beforeEach(() => {
+    tmpdir = getTmpDir();
+  });
+
+  afterEach(() => {
+    try {
+      fs.rmSync(tmpdir, { recursive: true, force: true });
+    } catch (err) {
+      // Ignore cleanup errors
+    }
+  });
+
+  it("returns first folder created with recursive when all folders are new", async () => {
+    const dir1 = nextdir();
+    const dir2 = nextdir();
+    const firstPathCreated = path.join(tmpdir, dir1);
+    const pathname = path.join(tmpdir, dir1, dir2);
+
+    return new Promise<void>((resolve, reject) => {
+      fs.mkdir(pathname, { recursive: true }, (err, result) => {
+        if (err) return reject(err);
+        expect(fs.existsSync(pathname)).toBe(true);
+        expect(fs.statSync(pathname).isDirectory()).toBe(true);
+        expect(result).toBe(path.toNamespacedPath(firstPathCreated));
+        resolve();
+      });
+    });
+  });
+
+  it("returns last folder created with recursive when only last folder is new", async () => {
+    const dir1 = nextdir();
+    const dir2 = nextdir();
+    const pathname = path.join(tmpdir, dir1, dir2);
+
+    // Create the parent directory
+    fs.mkdirSync(path.join(tmpdir, dir1));
+
+    return new Promise<void>((resolve, reject) => {
+      fs.mkdir(pathname, { recursive: true }, (err, result) => {
+        if (err) return reject(err);
+        expect(fs.existsSync(pathname)).toBe(true);
+        expect(fs.statSync(pathname).isDirectory()).toBe(true);
+        expect(result).toBe(path.toNamespacedPath(pathname));
+        resolve();
+      });
+    });
+  });
+
+  it("returns undefined with recursive when no new folders are created", async () => {
+    const dir1 = nextdir();
+    const dir2 = nextdir();
+    const pathname = path.join(tmpdir, dir1, dir2);
+
+    // Create the directories first
+    fs.mkdirSync(pathname, { recursive: true });
+
+    return new Promise<void>((resolve, reject) => {
```

No need to change but I normally prefer async/await or `done()/ done(err)` instead of returning a promise like this, and when I need a promise/resolve/reject is normally to be awaited inside the context (using Promise.withResolvers()) instead of chained.